siliconcompiler/dependencyschema.py

```diff
@@ -25,7 +29,38 @@ def __init__(self):
                 shorthelp="List of dependencies",
                 help="List of named object dependencies included via add_dep()."))
 
-        self.__deps = {}
+        schema.insert(
+            'source', 'default', 'path',
+            Parameter(
+                'str',
+                scope=Scope.GLOBAL,
+                shorthelp="Package: data source path",
+                example=[
+                    "api: chip.set('source', "
+                    "'freepdk45_data', 'path', 'ssh://git@github.com/siliconcompiler/freepdk45/')"],
+                help=trim("""
+                Package data source path, allowed paths:
+
+                * /path/on/network/drive
+                * file:///path/on/network/drive
+                * git+https://github.com/xyz/xyz
+                * git://github.com/xyz/xyz
+                * git+ssh://github.com/xyz/xyz
+                * ssh://github.com/xyz/xyz
+                * https://github.com/xyz/xyz/archive
+                * https://zeroasic.com/xyz.tar.gz
+                * python://siliconcompiler
+                """)))
+
+        schema.insert(
```

Comment:
Not clear from keys or example what this does. Looking at the example, this looks like a commit hash? 'ref' is a very anonymous word.

How about 'tag'? With an expanded help explanation that the meaning is context dependent on the package root and could refer to commit hash, tag, label, release, or version market,..

Reply:
tag works